google/cloud/storage/blob.py

```diff
@@ -2660,3 +2660,15 @@ def _raise_for_more_than_one_none(**kwargs):
         )
 
         raise ValueError(msg)
+
+
+def _insert_generation_match_args(name_value_pairs, **kwargs):
+    for snake_field, camel_field in {
+        "if_generation_match": "ifGenerationMatch",
+        "if_generation_not_match": "ifGenerationNotMatch",
+        "if_metageneration_match": "ifMetagenerationMatch",
+        "if_metageneration_not_match": "ifMetagenerationNotMatch",
+    }.items():
+        value = kwargs.get(snake_field)
+        if value is not None:
+            name_value_pairs.append((camel_field, value))
```

Comment:
We're taking generation match args as `kwargs` which is `dict`. Then we're walking through generation fields names checking if any of the given generation args are not `None`, and adding them into the list of other params. See usage example:

![Безымянный](https://user-images.githubusercontent.com/26030071/80463650-3e572700-8941-11ea-8899-0f03f2366b3c.png)

That'll be good to make this dict with snake case and camel case field names constant, not to create it again on every helper call.

It seems to me this way is better then adding into every method constructions like this:
https://github.com/q-logic/python-storage/blob/311ecabf8acc3018cef0697dd29483693f7722b9/google/cloud/storage/blob.py#L1076-L1088

Reply:
I'd say don't scope in `user_project`, `kms_key_name` and `predefined_acl`. 

Will you still surface parameters in the method signature? 

I think this looks reasonable to reduce duplication of code.

Reply:
> Will you still surface parameters in the method signature?

Well, I like using `**kwargs` instead of itemizing every `*generation*match` arg, but if to think very carefully, using `**kwargs` makes methods signature not very clear, so I think I'll write:

```python
def method(
    ...
    if_generation_match=None, 
    if_generation_not_match=None, 
    if_metageneration_match=None, 
    if_metageneration_not_match=None,
):
```
instead of

```python
def method(
    ...
    **kwargs
)
```
in public methods, and will be using `**kwargs` only in this helper, to reduce its code length.

Reply:
+1, please expose parameters otherwise it's going to be confusing.